### ISSUE TITLE
Unexpected k400 test results

### ISSUE THREAD

**Issue by adeobootpin** [OpenGVLab/UniFormerV2#42]
I ran the K400 UniFormerV2-B/16 model on the test set. For the CLIP-400M pretrained version, I got a top 1 accuracy of 82.10 and top 5 accuracy of 95.62. This did not surprise me even though it was below what I expected.

What I found strange was that when I ran the CLIP-400M+K710 version, I got a top 1 accuracy of 90.75 and a top 5 accuracy of 98.54! I expected ~85.6 top 1 accuracy at best. How is this possible? What did I do wrong?

In addition, I am also not sure whether to load the checkpoint or not (k400_k710_uniformerv2_b16_8x224.pyth). I get the above result when I do. When I don't load the checkpoint and use the pretrained weights (k710_uniformerv2_b16_8x224.pyth), I get at top 1 accuracy of 89.49 and a top 5 accuracy of 97.90.

**Comment by Andy1621** [comment]
It's strange for your experiments. Can you provide more details? Such as your label file of K400?

Our results have been verified in [MMAction2](https://github.com/open-mmlab/mmaction2/tree/dev-1.x/configs/recognition/uniformerv2). Note that due to different training and validation files of K600 and K700, the results have a gap.

**Comment by adeobootpin** [comment]
Hi, thanks for the prompt response.
Attached are my test.csv and the yaml files.

I set the following in the main() function:
    cfg.NUM_GPUS = 2
    cfg.TEST.CHECKPOINT_FILE_PATH = '/k400_k710_uniformerv2_b16_8x224.pyth'
    cfg.DATA_LOADER.NUM_WORKERS = 1
    cfg.TEST.BATCH_SIZE = 24
    cfg.DATA.PATH_TO_DATA_DIR = "/UniFormerV2"
    cfg.DATA.PATH_LABEL_SEPARATOR = ","
    cfg.TEST.ADD_SOFTMAX = True
    cfg.TEST.NUM_ENSEMBLE_VIEWS = 4 
    cfg.TEST.NUM_SPATIAL_CROPS = 3
    #cfg.TEST.TEST_BEST = True    
    cfg.RNG_SEED = 6666

I have tried running with cfg.TEST.CHECKPOINT_FILE_PATH set as well as without setting it. 

The command I use to run the test:
python3 tools/run_net_multi_node.py --cfg exp/k400/test_B_16.yaml

[test.csv](https://github.com/OpenGVLab/UniFormerV2/files/11315954/test.csv)
[test_B_16.yaml.txt](https://github.com/OpenGVLab/UniFormerV2/files/11315960/test_B_16.yaml.txt)

**Comment by adeobootpin** [comment]
Any idea what is responsible for my results? Did the k400 test set leak into your k710 training set?

**Comment by adeobootpin** [comment]
Hi, thanks for your response. From the link you shared, I notice that the test and validation csv files for k400 are identical. 

Is that intentional? 
If so, were the models used for comparison in your paper retested with this same test set?

**Comment by Andy1621** [comment]
Yes! The test and validation are identical. Since the test file do not release the ground truth, people often use validation file for testing.

**Comment by adeobootpin** [comment]
Hi, thanks for your response. 
The K400 test file does include ground truth (https://storage.googleapis.com/deepmind-media/Datasets/kinetics400.tar.gz) and so I assumed that was what was used for testing.

**Comment by Andy1621** [comment]
Just use the validation file!

**Comment by adeobootpin** [comment]
Interesting, so you added some of the examples from the test set to your training set.

**Comment by Andy1621** [comment]
what?

**Comment by Andy1621** [comment]
The train/validation split files I provided are commonly used versions. The test split is not a popular benchmark.

"Interesting, so you added some of the examples from the test set to your training set." The files I provided do not have overlap between training and validation.

**Comment by adeobootpin** [comment]
This is from the kinetics 400 data set paper (arXiv:1705.06950):

**Statistics: The dataset has 400 human action classes, with 400–1150 clips for each action, each from a unique video.
Each clip lasts around 10s. The current version has 306,245 videos, and is divided into three splits, one for training having 250–1000 videos per class, one for validation with 50 videos per class and one for testing with 100 videos per class**

Note: There are 3 splits, not 2. Training, validation and testing.

I am trying to compare models using published results and wondering if merging the training and test sets is common practice.

**Comment by Andy1621** [comment]
For your problem: "if merging the training and test sets is common practice". 
We do not merge the training and test data, but there is some overlap between the K400 test videos with K600 and K700 training videos. Again, the test set is not a popular benchmark in current research. The validation set is a relatively fair benchmark.  
Besides, we also report those results without K710 pertaining in our paper. We just propose K710 to explore whether large-scale supervised pertaining can achieve better results on K400/K600/K700 validation set, which do not have any overlap. A similar try can be found in JFT3B pretraining for ImageNet.

**Comment by adeobootpin** [comment]
OK fair enough. Thanks for the feedback.
